themes/rjorgenson/rjorgenson.theme.sh

```diff
@@ -14,88 +14,82 @@ SCM_THEME_PROMPT_CLEAN=" ${bold_green}✓${normal}"
-        TITLEBAR=""
-        ;;
+        xterm*) TITLEBAR="\[\033]0;\w\007\]" ;;
 esac
 
-PS3=">> "
+if [[ $OSTYPE =~ "darwin" ]]; then
+    echo "${BRACKET}➞  ${normal}"
+fi
 
 __my_rvm_ruby_version() {
-    local gemset=$(echo $GEM_HOME | awk -F'@' '{print $2}')
-  [ "$gemset" != "" ] && gemset="@$gemset"
-    local version=$(echo $MY_RUBY_HOME | awk -F'-' '{print $2}')
-    local full="$version$gemset"
-  [ "$full" != "" ] && echo "${BRACKET_COLOR}[${STRING_COLOR}$full${BRACKET_COLOR}]${normal}"
+    local gemset ; gemset=$(echo "${GEM_HOME}" | awk -F'@' '{print $2}')
+    local version ; version=$(echo "${MY_RUBY_HOME}" | awk -F'-' '{print $2}')
+    [ "${gemset}" != "" ] && gemset="@${gemset}"
+    local full="${version}${gemset}"
+    [ "${full}" != "" ] && echo "${BRACKET}[${STRING_COLOR}${full}${BRACKET}]${normal}"
 }
 
 is_vim_shell() {
-        if [ ! -z "$VIMRUNTIME" ]
-        then
-                echo "${BRACKET_COLOR}[${STRING_COLOR}vim shell${BRACKET_COLOR}]${normal}"
-        fi
+    if [ ! -z "$VIMRUNTIME" ]
+    then
+        echo "${BRACKET}[${STRING_COLOR}vim shell${BRACKET}]${normal}"
+    fi
 }
 
 function is_integer() { # helper function for todo-txt-count
     [ "$1" -eq "$1" ] > /dev/null 2>&1
         return $?
 }
 
+# XXX do we need/want to integrate with todo.sh? We don't provide it and I
+# can't find a version online that accepts ls as an input
 todo_txt_count() {
     if _omb_util_command_exists todo.sh; then # is todo.sh installed
-        count=$(todo.sh ls | egrep "TODO: [0-9]+ of ([0-9]+) tasks shown" | awk '{ print $4 }')
-        if is_integer $count; then # did we get a sane answer back
-            echo "${BRACKET_COLOR}[${STRING_COLOR}T:$count${BRACKET_COLOR}]$normal"
+        count=$(todo.sh ls \
+          | awk '/TODO: [0-9]+ of ([0-9]+) tasks shown/ { print $4 }')
+        if [ "${count}" -eq "${count}" ] &> /dev/null
+        then # did we get a sane answer back
+            echo "${BRACKET}[${STRING_COLOR}T:$count${BRACKET}]$normal"
         fi
     fi
 }
 
 modern_scm_prompt() {
-        CHAR=$(scm_char)
-        if [ $CHAR = $SCM_NONE_CHAR ]
-        then
-                return
-        else
-                echo "${BRACKET_COLOR}[${CHAR}${BRACKET_COLOR}][${STRING_COLOR}$(scm_prompt_info)${BRACKET_COLOR}]$normal"
-        fi
-}
-
-my_prompt_char() {
-    if [[ $OSTYPE =~ "darwin" ]]; then
-        echo "${BRACKET_COLOR}➞  ${normal}"
-    else
-        echo "${BRACKET_COLOR}➞ ${normal}"
-    fi
+    CHAR=$(scm_char)
+    [ "${CHAR}" = "${SCM_NONE_CHAR}" ] \
+        || echo "${BRACKET}[${CHAR}${BRACKET}][${STRING_COLOR}$(scm_prompt_info)${BRACKET}]$normal"
 }
 
 _omb_theme_PROMPT_COMMAND() {
+    my_host="${STRING_COLOR}\h${normal}";
+    my_user="${STRING_COLOR}\u${normal}";
+    my_path="${STRING_COLOR}\w${normal}";
 
-    my_ps_host="${STRING_COLOR}\h${normal}";
-    my_ps_user="${STRING_COLOR}\u${normal}";
-    my_ps_root="${bold_red}\u${normal}";
-    my_ps_path="${STRING_COLOR}\w${normal}";
```

Comment:
I appreciate these shorter names, but in addition, we should make them `local` to avoid conflicts with the global variables that might be used by the user. `line2` should also be made `local`.

Reply:
fixed